### ISSUE TITLE
Merging & moving worlds

### ISSUE THREAD

**Issue by SanderMertens** [SanderMertens/flecs#122]
**Describe the problem you are trying to solve.**
Being able to merge or move the data from one world into another would allow for the asynchronous loading of content. With these features, deserialized data can be stored in a temporary world that is populated asynchronously, and moved / merged into the real world once loading is complete.

Merging a world into another world copies content from the source world into the destination world while not modifying the source, and not deleting entities from the destination world (though data may be overwritten).

Moving a world into another world moves the data from the source world to the destination world, thereby invalidating the source world and deleting data in moved tables from the destination world.

Merging is useful when loading content asynchronously. Moving is useful when restoring the world to a previous state.

**Describe the solution you'd like**
Two functions, `ecs_move_world` and `ecs_merge_world` that accept a destination and a source world.

**Comment by SanderMertens** [comment]
Potential solution to merge/copy worlds from src to dst:

```c
// Move data (no memcpy). If dst table exists, it will be emptied first
ecs_move_table(ecs_world_t *dst, ecs_rows_t*);

// Copy table. If dst table exists, data will be deleted or merged depending on replaced
ecs_copy_table(ecs_world_t *dst, ecs_rows_t*, bool replace);
```

Alternatively, application could use bulk operations:
```c
// Copy tables with world_cell_X_Y from world to dst_world
ecs_type_t type_world_cell_X_Y = ecs_type_from_entity(world, world_cell_X_Y);
ecs_filter_t filter = {
    .include = type_world_cell_X_Y
};

ecs_filter_iter it = ecs_filter_iter(world, &filter);

while (ecs_filter_next(&it)) {
    ecs_rows_t *rows = &it.rows;
    ecs_type_t table_type = ecs_table_type(rows);
    int32_t column_count = ecs_vector_count(table_type);
    void** table_columns = ecs_os_alloca(void*, column_count);

    for (int i = 0; i < column_count; i ++) {
        table_columns[i] = ecs_table_column(rows, i);
    }

    ecs_bulk_new_w_type(dst_world, table_type, rows->count, table_columns);
}
```

**Comment by SanderMertens** [comment]
I ended up creating an example that shows how to merge data between worlds with the direct access API. The `ecs_move_table` / `ecs_copy_table` functions would have had to make too many assumptions (about entity id mapping, whether src & dst tables have the same columns etc) that the number of functions to support this could easily explode. Instead, the direct access API provides direct access to a table and its columns, which provide the basic building blocks for an API to build their own version of `move_table` or `copy_table` that fits the needs.
